### PR TITLE
Witsml depth data support

### PR DESCRIPTION
**Pull request by Himanshu-it**
As per the present code, depth data (double) can be posted but not stored in the DB via mqtt. The control returns back at the start of dao layer i.e in function:
public void handleTelemetryUploadRequest(PluginContext ctx, TenantId tenantId, RuleId ruleId, TelemetryUploadRequestForDepthRuleToPluginMsg msg)

in file: TelemetryRuleMsgHandler.java

1. The mqtt topic name is : v1/devices/me/depth/telemetry
2. depth data sample : [{"ds":2000.1,"values":{"temperature":21, "humidity":55.0, "active": false}}]

### CODE REVIEW

File: common/message/src/main/java/org/thingsboard/server/common/msg/core/BasicTelemetryUploadRequestForDepth.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import lombok.ToString;
+import org.thingsboard.server.common.data.kv.KvEntry;
+import org.thingsboard.server.common.msg.session.MsgType;
+
+@ToString
+public class BasicTelemetryUploadRequestForDepth extends BasicRequest implements TelemetryUploadRequestForDepth {
```
Comment (jayeshkapadnis) [inline common/message/src/main/java/org/thingsboard/server/common/msg/core/BasicTelemetryUploadRequestForDepth.java]: Does it make sense to rename it to BasicDepthTelemetryUploadRequest and Interface to DepthTelemetryUploadRequest ?

File: common/transport/src/main/java/org/thingsboard/server/common/transport/adaptor/JsonConverter.java [hunk 2]
```diff
@@ -54,6 +56,29 @@ public static TelemetryUploadRequest convertToTelemetry(JsonElement jsonObject,
         return request;
     }
 
+    // HMDC test telemetry depth.
+    public static TelemetryUploadRequestForDepth convertToTelemetryDepth(JsonElement jsonObject, int requestId) throws JsonSyntaxException {
+        BasicTelemetryUploadRequestForDepth request = new BasicTelemetryUploadRequestForDepth(requestId);
+        log.error("\n\n HMDC request class: " + request.getClass() + "\n\n");
+        log.error("\n\n HMDC request val " + request + "\n\n");
```
Comment (jayeshkapadnis) [inline common/transport/src/main/java/org/thingsboard/server/common/transport/adaptor/JsonConverter.java]: These loggers can't be error, if you want make it debug or trace level like 
`log.debug("HMDC request class: {}", request.getClass());`

File: common/transport/src/main/java/org/thingsboard/server/common/transport/adaptor/JsonConverter.java [hunk 3]
```diff
@@ -68,6 +93,16 @@ private static void parseObject(BasicTelemetryUploadRequest request, long system
         }
     }
 
+    // HMDC telemetry ds
+    private static void parseObjectDs(BasicTelemetryUploadRequestForDepth request, JsonElement jsonObject) {
```
Comment (jayeshkapadnis) [inline common/transport/src/main/java/org/thingsboard/server/common/transport/adaptor/JsonConverter.java]: Please give meaningful names to methods

File: extensions-core/src/main/java/org/thingsboard/server/extensions/core/filter/MsgTypeFilter.java [hunk 4]
```diff
@@ -58,6 +60,8 @@ public void init(MsgTypeFilterConfiguration configuration) {
     @Override
     public boolean filter(RuleContext ctx, ToDeviceActorMsg msg) {
         for (MsgType msgType : msgTypes) {
+            log.error("\nHMDC msgtype : " + msgTypes);
+            log.error("\nHMDC payload msg type : " + msg.getPayload().getMsgType());
```
Comment (jayeshkapadnis) [inline extensions-core/src/main/java/org/thingsboard/server/extensions/core/filter/MsgTypeFilter.java]: Again change log level.

File: extensions/extension-rest-api-call/src/main/resources/DeviceAttributeRestDescriptor.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (jayeshkapadnis) [inline extensions/extension-rest-api-call/src/main/resources/DeviceAttributeRestDescriptor.json]: Do we need this action as part of this story Or have you mistakenly committed this ?

Comment (Himanshu-it) [reply extensions/extension-rest-api-call/src/main/resources/DeviceAttributeRestDescriptor.json]: By mistake I have done that. It is not in the latest commit.

File: extensions-core/src/main/java/org/thingsboard/server/extensions/core/filter/MsgTypeFilter.java (hunk 4 above)
Comment (Himanshu-it) [reply extensions-core/src/main/java/org/thingsboard/server/extensions/core/filter/MsgTypeFilter.java]: Sure, will do that.

**Pull request merged**